powerstore/resource_protection_policy.go

```diff
@@ -215,6 +215,58 @@ func (r resourceProtectionPolicy) Read(ctx context.Context, req tfsdk.ReadResour
 
 // Updates the protection policy
 func (r resourceProtectionPolicy) Update(ctx context.Context, req tfsdk.UpdateResourceRequest, resp *tfsdk.UpdateResourceResponse) {
+	log.Printf("Started Update")
+
+	//Get plan values
+	var plan models.ProtectionPolicy
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	//Get current state
+	var state models.ProtectionPolicy
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	protectionPolicyUpdate := r.planToProtectionPolicyParam(plan)
```

Comment:
Is removal of existing snapshot/replication rule possible with this?

Reply:
yes